### PR TITLE
EY/MIN-447. Document RPC calls

### PR DESCRIPTION
**Pull request by Jekshmek**
**Description:**

Added proper documentation for all RPC calls in the project:
1. The docs should contain short description of the RPC
2. The doc should clearly state the type of the returned data - asset or usd, percentage etc.
3. The docs should be unified across the RPCs and have similar structure.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

**Review by ales-by — DISMISSED** [review]

File: pallets/controller/rpc/src/lib.rs [hunk 1]
```diff
@@ -87,35 +100,38 @@ pub trait ControllerRpcApi<BlockHash, AccountId> {
...
-	///  - `&self` :  Self reference
+	/// Parameters:
+	///  - `&self`: Self reference
 	///  - `account_id`: current account id.
 	///  - `underlying_asset_id`: current asset id
 	///  - `at` : Needed for runtime API use. Runtime API must always be called at a specific block.
 	///
 	/// Return:
-	/// - amount: account total borrow per asset.
+	/// - [`amount`](`BalanceInfo::amount`): aaccount total borrow per asset (tokens amount).
```
Comment (lut1k) [inline pallets/controller/rpc/src/lib.rs:134]:
```suggestion
	/// - [`amount`](`BalanceInfo::amount`): user borrow underlying in a specific liquidity pool (underlying assets amount).
```

File: pallets/controller/rpc/src/lib.rs [hunk 2]
```diff
@@ -125,6 +141,15 @@ pub trait ControllerRpcApi<BlockHash, AccountId> {
 	) -> Result<Option<BalanceInfo>>;
 
 	/// Returns user underlying asset balance for the pool
```
Comment (lut1k) [inline pallets/controller/rpc/src/lib.rs:143]:
```suggestion
	/// Returns actual supply underlying balance for user in specific liquidity pool based on fresh latest indexes.
```

File: pallets/controller/rpc/src/lib.rs [hunk 3]
```diff
@@ -87,35 +100,38 @@ pub trait ControllerRpcApi<BlockHash, AccountId> {
...
-	///  - `at` : Needed for runtime API use. Runtime API must always be called at a specific block.
+	///  - `at`: Needed for runtime API use. Runtime API must always be called at a specific block.
 	///
 	/// Return:
-	/// - amount: account total collateral converted to usd.
+	/// - [`amount`](`BalanceInfo::amount`): account total collateral converted to usd.
 	#[rpc(name = "controller_accountCollateral")]
 	fn get_user_total_collateral(&self, account_id: AccountId, at: Option<BlockHash>) -> Result<Option<BalanceInfo>>;
 
 	/// Returns actual borrow balance for user per asset based on fresh latest indexes.
```
Comment (lut1k) [inline pallets/controller/rpc/src/lib.rs:125]:
```suggestion
	/// Returns actual borrow underlying balance for user per asset based on fresh latest indexes.
```

File: pallets/controller/rpc/src/lib.rs [hunk 4]
```diff
@@ -125,6 +141,15 @@ pub trait ControllerRpcApi<BlockHash, AccountId> {
...
 	/// Returns user underlying asset balance for the pool
+	///
+	/// Parameters:
+	///  - `&self`: Self reference
+	///  - `account_id`: current account id.
+	///  - `pool_id`: target pool id.
+	///  - `at` : Needed for runtime API use. Runtime API must always be called at a specific block.
+	///
+	///  Return:
+	///  - [`amount`](`BalanceInfo::amount`): user balance converted to underlying asset.
```
Comment (lut1k) [inline pallets/controller/rpc/src/lib.rs:152]:
```suggestion
	///  - [`amount`](`BalanceInfo::amount`): user supply underlying in a specific liquidity pool (underlying assets amount).
```

File: pallets/controller/src/lib.rs [hunk 5]
```diff
@@ -220,12 +220,32 @@ pub mod module {
...
 	/// max_borrow_rate)`.
+	///
+	/// Return:
+	/// - `last_interest_accrued_block`: block number that interest was last accrued at
+	/// - `protocol_interest_factor`: defines the portion of borrower interest that is converted
+	/// into protocol interest
+	/// - `max_borrow_rate`:  Maximum Borrow Rate is used to block the protocol functioning,
+	/// if the rate goes higher than this value
+	/// - `collateral_factor`: Collateral factor determines how much a user can borrow based on
+	/// their collateral value
```
Comment (lut1k) [inline pallets/controller/src/lib.rs:231]:
```suggestion
	/// - `collateral_factor`: multiplier, represents which share of the supplied value can be used as a collateral for loans.
```

**Comment by lut1k** [comment]
I merged your branch with master, make a git pool before fixing comments. :)

File: pallets/liquidity-pools/src/lib.rs [hunk 6]
```diff
@@ -129,10 +129,27 @@ pub mod module {
 		PoolAlreadyCreated,
 	}
 
+	/// Return liquidity pools information: (borrowed, borrow_index, total_protocol_interest)
```
Comment (lut1k) [inline pallets/liquidity-pools/src/lib.rs:132]:
```suggestion
	/// Return liquidity pools information: (borrowed, borrow_index, protocol_interest)
```

File: pallets/liquidity-pools/src/lib.rs [hunk 7]
```diff
@@ -129,10 +129,27 @@ pub mod module {
...
+	/// - `borrow_index`: Borrow Index accumulates the total earned interest since the opening of
+	/// the pool.
+	/// Used to accrue interest when user repays a loan.
+	/// - `protocol_interest`: amount of protocol_interest of the underlying held in this pool.
 	#[pallet::storage]
 	#[pallet::getter(fn pool_data_storage)]
 	pub(crate) type PoolDataStorage<T: Config> = StorageMap<_, Twox64Concat, CurrencyId, PoolData, ValueQuery>;
 
+	/// Return information about the user of the liquidity pool: (borrowed, interest_index,
+	/// is_collateral, liquidation_attempts)
```
Comment (lut1k) [inline pallets/liquidity-pools/src/lib.rs:146]:
```suggestion
	/// Return information about the user of the liquidity pool: (borrowed, interest_index,
	/// is_collateral,)
```

File: pallets/minterest-model/src/lib.rs [hunk 8]
```diff
@@ -171,8 +181,10 @@ pub mod module {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Set JumpMultiplierPerBlock from JumpMultiplierPerYear.
-		/// - `pool_id`: PoolID for which the parameter value is being set.
-		/// - `jump_multiplier_rate_per_year`: used to calculate multiplier per block.
+		///
+		/// Parameters:
+		/// - `pool_id`: the CurrencyId of the pool for which the parameter value is being set.
+		/// - `jump_multiplier_rate_per_year`:  new jump multiplier rate per year value.
```
Comment (lut1k) [inline pallets/minterest-model/src/lib.rs:187]:
```suggestion
		/// - `jump_multiplier_rate_per_year`:  new jump multiplier rate per year value. Used to calculate and set up multiplier per block.
```

File: pallets/minterest-model/src/lib.rs [hunk 9]
```diff
@@ -206,8 +218,10 @@ pub mod module {
 		}
 
 		/// Set BaseRatePerBlock from BaseRatePerYear.
-		/// - `pool_id`: PoolID for which the parameter value is being set.
-		/// - `base_rate_per_year`: used to calculate rate per block.
+		///
+		/// Parameters:
+		/// - `pool_id`: the CurrencyId of the pool for which the parameter value is being set.
+		/// - `base_rate_per_year`: new base rate per year value.
```
Comment (lut1k) [inline pallets/minterest-model/src/lib.rs:224]:
```suggestion
		/// - `base_rate_per_year`: new base rate per year value. Used to calculate and set up base rate per block.
```

File: pallets/minterest-model/src/lib.rs [hunk 10]
```diff
@@ -248,8 +262,10 @@ pub mod module {
 		}
 
 		/// Set MultiplierPerBlock from MultiplierPerYear.
-		/// - `pool_id`: PoolID for which the parameter value is being set.
-		/// - `multiplier_per_year`: used to calculate multiplier per block.
+		///
+		/// Parameters:
+		/// - `pool_id`: the CurrencyId of the pool for which the parameter value is being set.
+		/// - `multiplier_per_year`: new multiplier per year value.
```
Comment (lut1k) [inline pallets/minterest-model/src/lib.rs:268]:
```suggestion
		/// - `multiplier_per_year`: new multiplier per year value. Used to calculate and set up multiplier per block.
```

**Review by lut1k — DISMISSED** [review]

File: pallets/minterest-protocol/src/lib.rs [hunk 11]
```diff
@@ -465,6 +484,9 @@ pub mod module {
 		}
 
 		/// Sender allowed the assets in the pool to be used as collateral.
+		///
+		/// Parameters:
+		/// `pool_id`: CurrencyId of a pool that  should be used a Collateral.
```
Comment (Apollo50) [inline pallets/minterest-protocol/src/lib.rs:489]:
```suggestion
		/// `pool_id`: CurrencyId of a pool that should be used a Collateral.
```

File: pallets/mnt-token/rpc/src/lib.rs [hunk 12]
```diff
@@ -32,11 +37,18 @@ pub trait MntTokenRpcApi<BlockHash, AccountId> {
...
+	/// Parameters:
 	///  - `&self` :  Self reference
 	///  - `pool_id`: current pool id.
 	///  - `at` : Needed for runtime API use. Runtime API must always be called at a specific block.
+	///
 	/// Return:
-	/// - (borrow_rate, supply_rate): MNT Borrow Rate and MNT Supply Rate values
+	///
+	///   (borrow_rate, supply_rate)
+	///
```
Comment (Apollo50) [inline pallets/mnt-token/rpc/src/lib.rs:49]:
```suggestion
```

File: pallets/whitelist/src/lib.rs [hunk 13]
```diff
@@ -164,7 +165,9 @@ pub mod module {
 
 		/// Remove a member from the whitelist.
 		///
+		/// Parameters:
 		/// - `who`: the account that is being removed from the whitelist.
+		/// - `account_to_remove`: AccountId of the removed member.
```
Comment (Apollo50) [inline pallets/whitelist/src/lib.rs:170]:
```suggestion
		/// - `account_to_remove`: the account that is being removed from the whitelist.
```

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]
Please, fix conflicts.

**Pull request merged**